File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {

        int CurrNum = 0;
        int MaxNum = 0;

        for(int i = 0; i < nums.length; i++)
        {
            if(nums[i] == 1)
            {
                CurrNum++;
            }
            else
            {
                MaxNum = Math.max(MaxNum, CurrNum);
                CurrNum = 0;
            }
        }

        MaxNum = Math.max(MaxNum, CurrNum);

        return MaxNum;
    }
}

File: 485-max-consecutive-ones/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let previous_count = 0;
    let current_count = 0;

    for(let i = 0; i<nums.length; i++)

    {
        if(nums[i] == 1)
        {
            previous_count++;
        }
        else
        {
               current_count = Math.max(previous_count,current_count);
               previous_count = 0;
        }
    
        
    }
    current_count = Math.max(previous_count,current_count);

    return current_count;
    
};